fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -497,10 +497,40 @@ protected static void addCacheControlHeaders(final HttpServletResponse servletRe
             return;
         }
 
-        final FedoraResource mutableResource = resource instanceof NonRdfSourceDescription
-                ? ((NonRdfSourceDescription) resource).getDescribedResource() : resource;
-        final EntityTag etag = new EntityTag(mutableResource.getEtagValue());
-        final Date date = mutableResource.getLastModifiedDate();
+        final EntityTag etag;
+        final Date date;
+
+        // The HTTP headers for ETags and Last-Modified dates are swapped for fedora:Binary
```

Comment:
This is all good info, but there's so much of it that maybe it should extracted to a Javadoc-level comment about the kinds of resources at play and the general considerations for each (e.g. LDP resources get triples for description, HTTP resources get headers, etc.).